src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -405,5 +405,16 @@ public void Dispose()
             State = FunctionInvocationDispatcherState.Disposing;
             Dispose(true);
         }
+
+        public async Task RestartAsync()
+        {
+            // Dispose and restart all initialized channels
+            State = FunctionInvocationDispatcherState.WorkerProcessRestarting;
+            var channels = await GetInitializedWorkerChannelsAsync();
```

Comment:
You need to restart all channels not just the ones that are 'Initialized'

Reply:
What would be a scenario where a channel is in "Initializing" or "Default" state when a timeout happens? If invocations are only happening through "Initialized" channels, doesn't it make sense to restart only those? Unless I'm missing something here...

Reply:
A worker channel is initialized after  worker startes and grpc channel is established...not restarting all the channels will lead to a race condition if jobhost restart is issued before grpc channel is initialized.

Reply:
@pragnagopa - i think i'm also confused here. I don't think there will be a jobhost restart on a timeout? Or are you talking about a jobhost restart for a different reason?

Reply:
@mhoeger - this fix is about restarting jobhost if function invocations timeout instead of shutting down whole application

Reply:
@pragnagopa - i think the changes only restart the worker channels, whether they are at jobhost or webhost scope (which makes sense - because functions and host configuration are not excepted to change on a timeout).

Reply:
The change is only around restarting worker channels (and therefore recycling the processes), and not an entire jobhost restart..

Reply:
My bad. I recommend JobHost restart which aligns well with holding off reading events from event sources until jobhost restart completes. This would be cleaner approach.

Reply:
Taking my comment back. Lets test the changes where restarting channels works :)

Reply:
I don't think that we wait for worker channels to be up and running to read events from event sources, so a JobHost restart seems like overkill? Although if we can change that in the future this wouldn't benefit from that change. (I ran a quick test to add a 20 second delay in sending worker init response to simulate slow startup:)
<img width="813" alt="screenshot" src="https://user-images.githubusercontent.com/8888531/76895422-3047be00-684d-11ea-8175-26f89aadefbd.png">

Reply:
I tested this out and noticed the process id change after recycling.
![image](https://user-images.githubusercontent.com/26885664/76896236-aef12b00-684e-11ea-87af-a44a7463ca7e.png)

Reply:
This just came up in a CRI analysis. App is using multiple worker processes - 10. Only invocation on one of them timeouted out, rest of the 9 processes are still healthy. What you and @mhoeger proposed makes more sense. It also, makes sense to only restart worker channels that are faulted due to timeout. So introduce, another state for channel 'Faulted' and only restart channels that are faulted.